src/dynamics/_common/sticky_handler.js

```diff
@@ -118,11 +119,12 @@ Scoped.define("module:StickyHandler", [
                             this.trigger("transitionOutOfView");
                             return;
                         }
+                        var r = this.element.getBoundingClientRect();
                         if (this.noFloatIfAbove || this.noFloatIfBelow) {
-                            var r = this.element.getBoundingClientRect();
                             if (this.noFloatIfAbove && r.top >= 0) return;
                             if (this.noFloatIfBelow && r.top <= 0) return;
                         }
+                        if (!this.noFloatIfAboveCondition(r)) return
```

Comment:
@maxblessed we should trigger transitionOutOfView if the player doesn't transition to float